src/software/pipeline/main_cameraInit.cpp

```diff
@@ -130,6 +131,18 @@ inline std::istream& operator>>(std::istream& in, EGroupCameraFallback& s)
     return in;
 }
 
+std::list<std::string>::iterator findColorProfile(std::pair<std::string, std::string>& p, std::list<std::string>& fileList)
+{
+    std::list<std::string>::iterator it = fileList.begin();
+    bool found = false;
+    while (it != fileList.end() && !found)
+    {
+        found = (it->find(p.first) != std::string::npos) && (it->find(p.second) != std::string::npos);
+        if (!found)
+            ++it;
+    }
+    return it;
+}
```

Comment:
you can use c++ `find_if()`, something like this:
```suggestion
{
   return std::find_if(begin(fileList), end(fileList), [p](const std::string& s)
    { 
        return (s.find(p.first) != std::string::npos) && (s.find(p.second) != std::string::npos); 
        
    });
}
```